server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -219,7 +219,9 @@ protected final void innerWriteTo(StreamOutput out, boolean asKey) throws IOExce
             out.writeOptionalString(preference);
         }
         if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
-            out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
+            if (asKey == false) {
+                out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
+            }
```

Comment:
@jimczi  I was wondering if should also do this for `bottomSortValues` as well?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/54071/commits/d7d1c86d5ccf9d130d83aaa83c8c74cb0cf63d57